action/action-util/src/main/java/com/powsybl/action/util/ProportionalScalable.java

```diff
@@ -25,30 +99,65 @@ private static void checkPercentages(List<Float> percentages, List<Scalable> sca
         }
         double sum = percentages.stream().mapToDouble(Double::valueOf).sum();
         if (Math.abs(100 - sum) > 0.01) {
-            throw new IllegalArgumentException("Sum of percentages must be equals to 100 (" + sum + ")");
+            throw new IllegalArgumentException(String.format("Sum of percentages must be equals to 100 (%.2f)", sum));
         }
     }
 
-    private final List<Float> percentages;
+    private boolean notSaturated() {
+        return scalablePercentageList.stream().anyMatch(ScalablePercentage::notSaturated);
+    }
 
-    ProportionalScalable(List<Float> percentages, List<Scalable> scalables) {
-        super(scalables);
-        checkPercentages(percentages, scalables);
-        this.percentages = Objects.requireNonNull(percentages);
+    private void checkIterationPercentages() {
+        double iterationPercentagesSum = scalablePercentageList.stream().mapToDouble(ScalablePercentage::getIterationPercentage).sum();
+        if (Math.abs(100 - iterationPercentagesSum) > 0.01) {
+            throw new AssertionError(String.format("Error in proportional scalable ventilation. Sum of percentages must be equals to 100 (%.2f)", iterationPercentagesSum));
```

Comment:
If it can happen because of bad user input it should not be an assertion error but an illegalargumentexception or powsyblexception.

Reply:
The bad input is checked in Constructor (with IllegalArgumentException). Here it is the result of iteration percentages update done in the scaling algorithm. For me it is an AssertionError.

Reply:
OK